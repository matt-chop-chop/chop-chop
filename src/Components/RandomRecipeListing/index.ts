export { default as RandomRecipeListing } from "./RandomRecipeListing";
