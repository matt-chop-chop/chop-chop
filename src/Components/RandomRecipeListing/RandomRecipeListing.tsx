import { useRandomRecipe } from "@/hooks";
import { ErrorCard, LoadingCard, RecipeListing } from "@/Components";
import React from "react";

export type RandomRecipeListingProps = {
  listingIndex?: number;
};

const RandomRecipeListing = ({
  listingIndex = 0,
}: RandomRecipeListingProps) => {
  const { recipe, loading, error } = useRandomRecipe(listingIndex);

  if (loading) {
    return <LoadingCard />;
  }

  if (error) {
    return <ErrorCard error={error} />;
  }

  if (!recipe) {
    return <ErrorCard error={new Error("Recipe could not be found")} />;
  }

  return <RecipeListing recipe={recipe} />;
};

export default RandomRecipeListing;
