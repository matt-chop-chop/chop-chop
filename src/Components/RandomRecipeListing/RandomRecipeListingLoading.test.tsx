import { shallow } from "enzyme";
import React from "react";
import RandomRecipeListing from "./RandomRecipeListing";

jest.mock("@/hooks/useRandomRecipe", () => ({
  __esModule: true,
  useRandomRecipe: () => ({ recipe: null, loading: true }),
}));

describe("RandomRecipeListing", () => {
  it("is exported from RecipeListing", () => {
    expect(RandomRecipeListing).toBeDefined();
  });

  it("renders as expected when loading", () => {
    const wrapper = shallow(<RandomRecipeListing />);

    expect(wrapper).toMatchSnapshot();
  });
});
