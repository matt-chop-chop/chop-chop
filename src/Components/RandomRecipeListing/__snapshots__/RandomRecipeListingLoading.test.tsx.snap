// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomRecipeListing renders as expected when loading 1`] = `<LoadingCard />`;
